rabbitmq/src/main/java/io/micronaut/rabbitmq/connect/DefaultChannelPool.java

```diff
@@ -82,6 +82,13 @@ public Channel getChannel() throws IOException {
             channel = channels.poll();
             if (channel == null) {
                 channel = createChannel();
+                // Check that the channel was actually created to avoid infinite loop
```

Comment:
Based on the Javadoc of `ChannelPool::getChannel` an `IOException` is more correct. 

> If a channel needed to be created and encountered an error